evals.py

```diff
@@ -0,0 +1,175 @@
+        cfg.dataset.sequence_size = train_cfg.dataset.sequence_size
+        val_data = load_cached_dataset(cfg.dataset, split=cfg.dataset_split)
+
+    dataloader = DataLoader(val_data, batch_size=train_cfg.train.batch_size)
+
+    input_size = len(val_data.src_vocab)
+    output_size = len(val_data.tgt_vocab)
+
+    model = make_model(
+        input_size=input_size,
+        output_size=output_size,
+        n=train_cfg.model.n,
+        d_model=train_cfg.model.d_model,
+        d_ff=train_cfg.model.d_ff,
+        dropout=train_cfg.model.dropout,
+    )
+    model.to(cfg.device)
+    model.load_state_dict(checkpoint["model_state_dict"])
+
+    if cfg.tasks.val_epoch:
+        pad_idx = val_data.tgt_vocab.index("<blank>")
+
+        criterion = LabelSmoothing(
+            size=output_size,
+            padding_idx=pad_idx,
+            smoothing=train_cfg.train.label_smoothing,
+        )
+        criterion.to(cfg.device)
+
+        print("Evaluating model ...")
+        loss, dist = val_epoch(
+            dataloader=dataloader,
+            model=model,
+            criterion=criterion,
+            device=cfg.device,
+        )
+        print(f"Model loss:   {loss} | Average distance:    {dist}")
+
+    if cfg.tasks.translation:
+        print("Checking model outputs ...")
+        translations = make_examples(
+            dataset=val_data,
+            model=model,
+            n_examples=cfg.n_examples,
+            random=cfg.random,
+            device=cfg.device,
+        )
+
+        for translation in translations:
+            print("Source (Input)        : " + " ".join(translation["src"]))
+            print("Target (Ground Truth) : " + " ".join(translation["tgt"]))
+            print("Model Output               : " + " ".join(translation["out"]))
+
+
+def load_checkpoint(run_name: str, epoch: str = "final", device: str = "cpu"):
+    # find path with desired run
+    path = None
+    for file in os.listdir("models"):
+        if file.find(f"{run_name}-{epoch}") > 0:
+            path = file
+            break
+
+    if path is None:
+        print("no run with this id found")
+        return None
+
+    path = "models/" + path
+    checkpoint = torch.load(path, map_location=device)
+    return checkpoint
+
+
+def make_examples(
+    dataset: BinsToVelocityDataset,
+    model: nn.Module,
+    start_index: int = 0,
+    n_examples: int = 5,
+    eos_string: str = "</s>",
+    random: bool = False,
+    device: str = "cpu",
+):
+    results = []
+    pad_idx = dataset.tgt_vocab.index("<blank>")
+    dataloader = DataLoader(dataset, shuffle=random)
+    idx = -1
+    for b in dataloader:
+        batch = Batch(b[0], b[1], pad_idx)
+        for it in range(len(batch)):
+            idx += 1
+            # I want to be able to get samples from any index from the database,
+            # I also want to use examples without overlap
+            if idx < start_index or idx % 2 == 1:
+                continue
+
```

Comment:
Like I hinted in other comments, I think sample selection should be separated from model output generation